File: BotShot/Devices/DriveBase.cs
using System;
using Microsoft.SPOT;

using CTRE.Phoenix;
using CTRE.Phoenix.MotorControl;
using CTRE.Phoenix.Controller;
using CTRE.Phoenix.MotorControl.CAN;
using CTRE.Phoenix.Sensors;

using BotShot.Utility;
using BotShot.Config;

namespace BotShot.Devices {
	public class DriveBase {
		// Objects associated with the drive base
		private TalonSRX motorL = new TalonSRX(DeviceIDs.DriveL);
		private TalonSRX motorR = new TalonSRX(DeviceIDs.DriveR);

		// constructor
		public DriveBase()
		{
			motorL.ConfigAllSettings(Motors.DriveL());
			motorR.ConfigAllSettings(Motors.DriveR());
		}

		//=== Functionality ==============================

		//================================================

		public void SetLeftPercent(float percentOutput)
		{
			// limit values
			if (percentOutput < -1.0f)
				percentOutput = -1.0f;
			else if (percentOutput > 1.0f)
				percentOutput = 1.0f;

			motorL.Set(ControlMode.PercentOutput, percentOutput);
		}

		public void SetRightPercent(float percentOutput)
		{
			// limit values
			if (percentOutput < -1.0f)
				percentOutput = -1.0f;
			else if (percentOutput > 1.0f)
				percentOutput = 1.0f;

			motorR.Set(ControlMode.PercentOutput, percentOutput);
		}

		public void Stop()
		{
			motorL.Set(ControlMode.PercentOutput, 0);
			motorR.Set(ControlMode.PercentOutput, 0);
		}
	}
}

File: DriveBase.cpp
#include "DriveBase.h"

TalonSRX DriveBase::_motorL = {DeviceIDs::driveL};
TalonSRX DriveBase::_motorR = {DeviceIDs::driveR};

void DriveBase::setLeftPercent(float percentOutput)
{
	#ifndef DRIVE
		return;
	#endif // DRIVE

	// limit values
	if (percentOutput < -1.0f)
		percentOutput = -1.0f;
	else if (percentOutput > 1.0f)
		percentOutput = 1.0f;

	_motorL.Set(ControlMode::PercentOutput, percentOutput);
}

void DriveBase::setRightPercent(float percentOutput)
{
	#ifndef DRIVE
		return;
	#endif // DRIVE
	
	// limit values
	if (percentOutput < -1.0f)
		percentOutput = -1.0f;
	else if (percentOutput > 1.0f)
		percentOutput = 1.0f;

	_motorR.Set(ControlMode::PercentOutput, -1 * percentOutput);
}

void DriveBase::stop()
{
	_motorL.Set(ControlMode::PercentOutput, 0.0);
	_motorR.Set(ControlMode::PercentOutput, 0.0);
}